R/Google.R

```diff
@@ -0,0 +1,207 @@
+#'  works the same as other national data sources, however, data from
+#'  Google supports three subregions (country, subregion and subregion2) which
+#'  can be accessed using the 'level' argument. There is also more data
+#'  available, such as hospitalisations data. The raw data comes as three
+#'  seperate data sets, "epidemiology" which is comprised of cases, tests and
+#'  deaths, "index", which holds information about countries linking the other
+#'  data sets, and "hospitalizations" which holds data about number of people
+#'  in hospital, ICU, etc.
+#'
+#' @source \url{https://github.com/GoogleCloudPlatform/covid-19-open-data}
+#' @export
+#' @concept dataset
+#' @examples
+#' # nolint start
+#' \dontrun{
+#' # get all countries
+#' national <- Google$new(level = "1", verbose = TRUE, steps = TRUE, get = TRUE)
+#' national$return()
+#' # show available countries
+#' national$show_counties()
+#' # get all regional data for the UK
+#' uk <- Google$new(regions = "uk", level = "2", verbose = TRUE, steps = TRUE, get = TRUE)
+#' uk$return()
+#' # get all subregional data for the UK
+#' uk <- Google$new(regions = "uk", level = "3", verbose = TRUE, steps = TRUE, get = TRUE)
+#' uk$return()
+#' }
+#' # nolint end
+Google <- R6::R6Class("Google",
+  inherit = CountryDataClass,
+  public = list(
+
+    #' @field origin name of country to fetch data for
+    origin = "Google",
+    #' @field supported_levels A list of supported levels.
+    supported_levels = list("1", "2", "3"),
+    #' @field supported_region_names A list of region names in order of level.
+    supported_region_names = list(
+      "1" = "country",
+      "2" = "subregion",
+      "3" = "subregion2"
+    ),
+    #' @field supported_region_codes A list of region codes in order of level.
+    supported_region_codes = list(
+      "1" = "iso_3166_1_alpha_3", "2" = "iso_code", "3" = "subregion2_code"
+    ),
+    #' @field common_data_urls List of named links to raw data.
+    # nolint start
+    common_data_urls = list(
+      "epidemiology" = "https://storage.googleapis.com/covid19-open-data/v2/epidemiology.csv",
+      "hospitalizations" = "https://storage.googleapis.com/covid19-open-data/v2/hospitalizations.csv",
+      "index" = "https://storage.googleapis.com/covid19-open-data/v2/index.csv"
+    ),
+    # nolint end
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c(
+      "new_confirmed",
+      "new_deceased",
+      "new_recovered",
+      "new_tested",
+      "total_confirmed",
+      "total_deceased",
+      "total_recovered",
+      "total_tested"
+    ),
+    #' @field country_info tibble of countries in the processed data
+    country_info = NULL,
+
+
+    #' @description Google specific download, calls `DataClass` download but
+    #' also fills the country info field
+    download = function() {
+      super$download()
+      if (!(is.na(self$supported_region_names[["1"]]))) {
```

Comment:
I don't this actually works for working out where has level 2 and level 3 data does it (which is what we need it for) it just returns a list of all countries which doesn't really help much?